CONTRIBUTING.md

```diff
@@ -131,7 +201,39 @@ how to deploy a test version of the schema documentation. This requires some bas
 
 ### Making Releases
 
-TODO: Add to this section later
+The release process for nmdc-schema is documented in the [NMDC Infrastructure Administration repository](https://github.com/microbiomedata/infra-admin/blob/main/releases/nmdc-schema.md).
+
+**Important:** Generated files in `nmdc_schema/` (such as `nmdc.py`, `nmdc-pydantic.py`, `nmdc_materialized_patterns.yaml`) should only be committed during the release process, not in regular PRs. These files are regenerated from the source schema during releases.
+
+### CLI Script Development Guidelines
+
+When creating or modifying CLI scripts for nmdc-schema, follow these conventions:
+
+**Script Structure:**
+- Scripts should use Click for CLI interfaces
+- Click parameters/options should use hyphens (e.g., `--input-file`, not `--input_file`)
+- Click parameters/options should use consistent naming across scripts
+- Entry points should typically be `main` or `cli` unless there's a compelling reason otherwise
+
+**Naming Conventions:**
+- Python file names: Use underscores as delimiters (e.g., `analyze_mentions_of_ids.py`)
+- CLI alias names: Use hyphens as delimiters (e.g., `analyze-mentions-of-ids`)
+- Input/output file names: Use hyphens as delimiters (e.g., `schema-output.yaml`)
+
+**Integration:**
+- All scripts must be registered in `[tool.poetry.scripts]` in `pyproject.toml`
```

Comment:
I would disagree with this. I don't see the point in adding internal-only scripts to `tool.poetry.scripts`.